applications/main/nfc/views/dict_attack.c

```diff
@@ -235,3 +249,28 @@ void dict_attack_inc_current_dict_key(DictAttack* dict_attack, uint16_t keys_tri
         },
         true);
 }
+
+void dict_attack_set_key_attack(DictAttack* dict_attack, bool is_key_attack) {
+    furi_assert(dict_attack);
+    with_view_model(
+        dict_attack->view,
+        DictAttackViewModel * model,
+        {
+            model->is_key_attack = is_key_attack;
+            model->key_attack_current_sector = 1;
```

Comment:
You start key attack not always from 1 sector

Reply:
Fixed